docs/management/gcache-write-set-cache-encryption.md

```diff
@@ -0,0 +1,100 @@
+# GCache and Write-Set cache encryption
+
+The feature is a [tech preview](../glossary.md#tech-preview). Before using this feature in production, we recommend that you test restoring production from physical backups in your environment, and also use the alternative backup method for redundancy.
+
+## GCache encryption Master Key
+
+GCache encrypts with a File Key. Creating the galera.gcache file or a full reset generates the File Key. The File Key is encrypted with the Master Key. 
+
+!!! admonition "See also"
+
+    For more information, see [Understanding GCache and Record-set Cache](../manual/gcache_record-set_cache_difference.md), and the Percona Database Performance Blog: [All you need to know about GCache](https://www.percona.com/blog/2016/11/16/all-you-need-to-know-about-gcache-galera-cache/)
+
+GCache has a RingBuffer on-disk file to manage write-sets. The file is pre-created when the server starts. This file contains a 1KB preamble to hold key-value pairs. The Master Key id and the encrypted File Key are stored in the GCache Ring preamble. The preamble is not encrypted because this section does not store sensitive data. 
+
+??? example "Sample preamble key-value pairs"
+
+    ```text
+
+    Version: 2
+    GID: 3afaa71d-6665-11ed-98de-2aba4aabc65e
+    synced: 0
+    enc_version: 1
+    enc_encrypted: 1
+    enc_mk_id: 3
+    enc_mk_const_id: 3ad045a2-6665-11ed-a49d-cb7b9d88753f
+    enc_mk_uuid: 3ad04c8e-6665-11ed-a947-c7e346da147f
+    enc_fk_id: S4hRiibUje4v5GSQ7a+uuS6NBBX9+230nsPHeAXH43k=
+    enc_crc: 279433530
+
+    ```
+
+### Key descriptions
+
+The following table describes the encryption keys defined in the preamble. All other keys in the preamble are not related to encryption.
+
+| Key | Description |
+|---|---|
+| `enc_version` | The encryption version |
+| `enc_encrypted` | If the GCache is encrypted or not |
+| `enc_mk_id` | A part of the Master Key ID. Rotating the Master Key increments the sequence number. |
+| `enc_mk_const_id` | A part of the Master Key ID, a constant Universally unique identifier (UUID). This option remains constant for the duration of the galera.gcache file. This option simplifies matching the keys inside the keyring to the instance that generated the keys. Deleting the galera.gcache changes this option. |
+| `enc_mk_uuid` | The first Master Key or if Galera detects that the preamble is inconsistent, which causes a full GCache reset and a new Master Key is required, generates this UUID. This change avoids reuse of existing keys, which may be compromised, from the keyring.  |
```

Comment:
`This change avoids reuse of existing keys, which may be compromised, from the keyring.`
Implementation detail. Remove

Reply:
Removed.